app/controllers/documents_controller.rb

```diff
@@ -59,7 +59,19 @@ def start; end
 
   def new_document
     document = Document.new(document_params.merge(chunking_profile: @chunking_profile))
-    document.filename = params[:file].original_filename
+    if params[:link].present?
+      begin
+        f = Tempfile.create(['web-', '.html'])
+        f.puts HTTParty.get(params[:link])
+        f.rewind
+        document.file = f
+        document.filename = File.basename(f.path)
```

Comment:
The tempfile names aren't very descriptive. What do you think about grabbing the page title and using that as the document filename?

Only the `file` attribute is managed by ActiveStorage; I don't think `file` and `filename` have to match.

Reply:
I tried that but it was kinda clunky and a html file on disk can have both a filename and title, so I'd rather add `title` to the documents table and display that instead of filename if it is not blank, if that's ok.